src/ports/postgres/modules/utilities/minibatch_preprocessing.py_in

```diff
@@ -363,21 +365,70 @@ class MiniBatchPreProcessorDL(MiniBatchPreProcessor):
 
         self._validate_args()
         self.num_of_buffers = self._get_num_buffers()
-        self.to_one_hot_encode = True
```

Comment:
I think we should document our motivation for always one hot encoding the dependent variable? I would say as a comment here in the code and also in the docs

Reply:
Also I think we should document the fact that our implementation of one hot encoding the dep var is consistent with the output of keras.to_categorical. IMO, we should document this in our code as well as the docs.

Reply:
Our 1-hot encoding follows the standard one-hot encoding convention. In fact, it is different from `keras.to_categorical`. For example, if there are 3 distinct class values captured in a list `y=[10, 11, 12]`, then the 1-hot encoded vector created by`keras.to_categorical(y)` is of size 13 (largest class value + 1). If it is called with `keras.to_categorical(y, num_classes=4)`, it errors out.
The 1-hot encoding done in MADlib would create a 1-hot encoded vector of size 4 in both cases.

I would say keras' 1-hot encoding is actually not the standard way of doing it.